src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -941,24 +941,39 @@ private boolean isChangeInteresting(Change change, GerritProject project, Gerrit
     }
 
     /**
-     * Should we trigger on this topic?
+     * Should we consider this topic w.r.t Topic Association options?
      *
      * @param topic the topic
      * @param project the configured gerrit project
+     * @param event the event
      * @return true if we should.
      */
-    private boolean isTopicInteresting(Topic topic, GerritProject project) {
-        for (GerritServer server : PluginImpl.getServers_()) {
-            logger.trace("query topic {} from {} ({}:{})", topic, server.getName(), server.getHostName(),
-                    server.getSshPort());
-            Map<Change, PatchSet> changes = topic.getChanges(server.getQueryHandler());
-            logger.trace("found {} changes with topic {} from {}", changes.size(), topic, server.getName());
-            for (Change change : changes.keySet()) {
-                if (isChangeInteresting(change, project, server.getQueryHandler())) {
-                    return true;
-                }
+    private boolean isTopicInteresting(Topic topic, GerritProject project, ChangeBasedEvent event) {
+
+        GerritServer server = PluginImpl.getServer_(event);
+
+        if (server == null) {
+            return false;
+        }
+
+        Map<Change, PatchSet> changes = topic.getChanges(server.getQueryHandler());
+        for (Change change : changes.keySet()) {
+
```

Comment:
keeping some `trace` or `debug` logging here would be nice I think if it is not behaving as expected for someone.